pennylane/measurements/expval.py

```diff
@@ -56,6 +56,12 @@ def circuit(x):
     Returns:
         ExpectationMP: measurement process instance
     """
+    if isinstance(op, qml.Identity) and len(op.wires) == 0:
+        # temporary solution to merge https://github.com/PennyLaneAI/pennylane/pull/5106
+        raise NotImplementedError(
+            "Expectation values of qml.Identity() without wires are currently not allowed."
+        )
+
```

Comment:
So one can measure `qml.expval(1. * qml.I())` but not `qml.expval(qml.I())`? Couldnt we just do

```suggestion
    if isinstance(op, qml.Identity) and len(op.wires) == 0:
        return ExpectationMP(obs=1.*op)

```
perhaps?

Reply:
So this was just me accidentally removing this logic and then re-adding it in a different place.  

While we now mostly support `ExpectationMP(qml.I())`, I think we should still have the error in `qml.expval` as a guard against something that might be more problematic.  We could even consider expanding the error to cover `qml.expval(2*qml.I())`.


Mostly works now, but I think we should probably do more validation and checking before making it available to a generic user.

Reply:
plugins. and everything else. As this PR shows, a lot of places weren't equipped to handle it.